docs/low-latency-v2-design.md

```diff
@@ -0,0 +1,1559 @@
+
+- **Connectivity Issues**: Collators having trouble connecting to the backing
+  group
+- **Relay chain forks**
+- **Overlapping parachain slots**
+- **Session Boundaries**: The relay chain is dropping candidates that are still
+  pending availability at session boundaries
+- **Relay chain block going overweight** (candidates get dropped)—most likely to
+  happen on parachain runtime upgrades
+- **Configuration Changes**: 
+  1. Very rare
+  2. Only causing problems, if the new configuration is not compatible with the
+     old one
+- **Parachain runtime upgrades**: Upgrade restriction signal is not properly
+  handled.
+
+#### Malicious Actors
+
+- **Collator Equivocation**—no consequences
+- **Backing group signing invalid blocks**—full slash of backers
+- **Backing group censoring** (blocks go out of scope)—less rewards for backers
+- **Relay chain block author censoring the parachain**—on purpose not backing
+  some candidates on chain
+
+### Problem Summary
+
+| Problem | Impact | Severity | Status |
+|---------|--------|----------|--------|
+| **Relay Chain Forks** | Blocks become obsolete | High | Inherent to current design |
+| **Collator Equivocation** | Double-spend attacks possible | High | No consequences currently |
+| **Session Boundaries** | Blocks dropped if pending at session change | High | Implementation issue (needs fix) |
+| **Relay Parent Expiry** | Blocks discarded if not included quickly | Medium | Limited validity window |
+| **Configuration Changes** | Blocks rejected due to config mismatch | Low | Rare, needs fix for edge cases |
+| **Runtime Upgrade Restriction** | Signal timing issues | Low | Edge case, needs fix |
+
+---
+
+## Goals
+
+We would like to achieve:
+
+1. **Much lower latency**. Instead of 36s, ideally we could get at least
+   somewhat secured confirmations within subseconds.
+
+2. **No single trusted party**. We won't depend on a single party to be honest.
+   No centralized "trusted" sequencer. Misbehavior cannot be prevented, but
+   requires collusion between collators and will be punished. Low-latency
+   confirmations won't be as strong as relay chain finality, but will still
+   derive some security from the relay chain: While the relay chain cannot
+   prevent any misbehavior, we will utilize it to ensure misbehavior will be
+   avenged, assuming at least one honest and live collator exists.
+
+3. **Opt-in add-on**. Low-latency confirmations are an opt-in add-on, not
+   affecting our current trust model for any user who chooses to rely on relay
+   chain finality. In particular censorship resistance and liveness properties
+   should be maintained.
+
+4. **High confidence in practice**. Low-latency should have high confidence in
+   practice even with malicious actors. While you should still wait for relay
+   chain finality for any high-value transactions, the day-to-day should work
+   with low-latency. We aim to be able to get collators slashed: So in essence
```

Comment:
Didn't yet get to the point where slashes are described but if we do want to have that, we should ensure that collators are more protected from DoS, like if using babe and having much greater numbers of collators.